### ISSUE TITLE
Can you use this for tiktok as well?

### ISSUE THREAD

**Issue by spinningsand** [go-olive/olive#12]
I noticed it worked on douyin and that you had tiktok tagged, but I wasn't sure.

**Comment by luxcgo** [comment]
Yes. Tiktok is fully supported in olive v0.2.9. Try for yourself!

**Comment by spinningsand** [comment]
Thanks for that! But I'm not exactly sure how to use it. Can you help me?

**Comment by luxcgo** [comment]
You can first download the executable file for your os platform. And change the config.toml file for your streamer.
Last simply run the command `/path/to/olive -c /path/to/config.toml`

**Comment by luxcgo** [comment]
1. `go install github.com/go-olive/olive/src/cmd/olive@latest`
2. `olive -c /path/to/config.toml`

how about trying this?

**Comment by luxcgo** [comment]
Actually, you have to download exe from https://github.com/go-olive/olive/releases. And put it somewhere, then cd that directory.

**Comment by spinningsand** [comment]
What's the difference between the to amd64? I noticed there are 2 different sizes for them.

**Comment by luxcgo** [comment]
```toml
[[Shows]]
Platform = "tiktok"
RoomID = "rpd.beans"
OutTmpl = "[{{ now | date \"2006-01-02 15-04-05\"}}][{{ .StreamerName }}]"
```

One block like this means one streamer, you can add or delete whatever streamers you want.

**Comment by spinningsand** [comment]
Awesome thanks! It downloaded a stream of someone I put in! One last question. Is it okay to leave it open when they're offline, would it start recording again if they go back online?

**Comment by luxcgo** [comment]
Of course, that's the key point.

**Comment by spinningsand** [comment]
Actually, I have 1 last question. When they're not live it outputs this every time it make a snapshot:

_←[31mERRO←[0m[2022-05-30 09:44:35] snap failed, User might be offline, Room ID not found  ←[31mid←[0m=promobot.robots←[31mpf←[0m=tiktok_

Is there a way you can make it where it doesn't output that as much and only when they're online? I can deal with it needs to stay that way, but I was just curious if you could do something about it so it'd look less cluttered.

**Comment by luxcgo** [comment]
Yes, I've just published a new release. u could use [olive v0.2.10](https://github.com/go-olive/olive/releases/tag/v0.2.10)

**Comment by spinningsand** [comment]
TikTok blocked my IP. Are there proxies or something that's supposed to be used with it?

**Comment by spinningsand** [comment]
I added about 4 different streamers to it and left it running for several hours while I went out and when I got back home, it was returning this:

_←[31mERRO←[0m[2022-05-30 17:20:19] snap failed, Your IP or country might be blocked by TikTok._

When I try to go on tiktok, it's a blank white screen, but I can still log into my account but only with a VPN.

**Comment by luxcgo** [comment]
Such conditions happened on other platform (kuaishou, a china tiktok competitor) as well. We might need to change a way of getting streamer infos which need some diggings.

**Comment by spinningsand** [comment]
Ahh okay. Could taking snapshots less often be a temporary fix until then? Like maybe if one was taken once every few minutes, it might be able to avoid getting the IP blocked.

**Comment by luxcgo** [comment]
Worth a try, I'll add the time interval as a configuration later.

**Comment by spinningsand** [comment]
Thanks, appreciate it.

**Comment by luxcgo** [comment]
feat is supported in v0.2.11, the param name is `SnapRestSeconds`

**Comment by spinningsand** [comment]
I changed the `SnaprestSeconds` value in config.toml and it doesn't do anything, it still takes a snap a snap every 5 seconds. I changed to 300 (for 5 minutes)

**Comment by luxcgo** [comment]
It works like a charm for me locally. Could u upload the config file?
And have u downloaded the new release? https://github.com/go-olive/olive/releases/tag/v0.2.11
And the name is not quite right.

Change the params in config file like the following and try again? 
```
LogLevel = 6
SnapRestSeconds = 300
```

**Comment by spinningsand** [comment]
Yes, I installed the newest version and when I try ,it just keeps taking snaps with little to no pause. It outputs this every few seconds:

_←[31mERRO←[0m[2022-06-01 20:38:44] snap failed, User might be offline, Room ID not found  ←[31mid←[0m=promobot.robots ←[31mpf←[0m=tiktok_

 It's only supposed to say that once every 5 minutes if you set the `SnapRestSeconds` param to 300, right? I'm not sure what `LogLevel` does but I set it to 6 as well. I barely had it running and it was already taking multiple snaps. This is what a I have in my config file, I have it to just 1 streamer now to reduce clutter until I can get it to work right:

```
LogLevel = 6
SnapRestSeconds = 300


[[Shows]]
Platform = "tiktok"
RoomID _=_ "promobot.robots"
OutTmpl = "[{{ now | date \"2006-01-02 15-04-05\"}}][{{ .StreamerName _}}]
```

**Comment by luxcgo** [comment]
If you change the `LogLevel` to 0 and rerun it, it's supposed to output nothing.
If not probably you're not using the latest version.

**Comment by luxcgo** [comment]
`LogLevel` controls the content that logger can log. It ranges from 0 to 6, as it grows larger, more detailed the logs become.

**Comment by spinningsand** [comment]
Since I've changed the `LogLevel` to 600 it hasn't blocked my IP, but one thing I've noticed is after a live ends, I get this:

_panic: close of closed channel

goroutine 9309 [running]:
github.com/Davincible/gotiktoklive.(*TikTok).newLive.func1()
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/live.go:60 +0x4b
github.com/Davincible/gotiktoklive.(*Live).parseWssMsg.func1()
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/wss.go:228 +0x44
created by github.com/Davincible/gotiktoklive.(*Live).parseWssMsg
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/wss.go:225 +0x5ad_

Then it shuts down.

**Comment by spinningsand** [comment]
Here's another output I get when the same thing happens:

`panic: Failed to read websocket message, atempting to reconnect: EOF

goroutine 293 [running]:
github.com/Davincible/gotiktoklive.routineErrHandler({0xc0016bd9c0?, 0x3c?, 0xc000249ec8?})
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/utils.go:330 +0x2d
github.com/Davincible/gotiktoklive.(*Live).readSocket(0xc0000e8d80)
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/wss.go:121 +0x305
created by github.com/Davincible/gotiktoklive.(*Live).tryConnectionUpgrade
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/wss.go:294 +0x136`

**Comment by luxcgo** [comment]
try [v0.2.12](https://github.com/go-olive/olive/releases/tag/v0.2.12), see if it happens.

**Comment by spinningsand** [comment]
It handles some lives for a while, but then it gave this output before it closed down again:

```
goroutine 2150 [running]:
github.com/Davincible/gotiktoklive.routineErrHandler({0xc0021874b0?, 0x3c?, 0xc00178fec8?})
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/utils.go:330 +0x2d
github.com/Davincible/gotiktoklive.(*Live).readSocket(0xc001c5e000)
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/wss.go:121 +0x305
created by github.com/Davincible/gotiktoklive.(*Live).tryConnectionUpgrade
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/wss.go:294 +0x136
```

**Comment by luxcgo** [comment]
[v0.2.13](https://github.com/go-olive/olive/releases/tag/v0.2.13) has fixed the bug, please try again.

**Comment by spinningsand** [comment]
Seems to be working good for now! I'll give another update if there are more issues. Thanks for all your help!

**Comment by spinningsand** [comment]
There is another thing I've noticed. Sometimes it only gets the audio while the video is a blank black screen.

**Comment by spinningsand** [comment]
The panic issue happened again as well. I noticed it only happens in lives when there's a likes battle. After it ended, it gave this output before shutting down: 

```
panic: Failed to read websocket message, atempting to reconnect: EOF

goroutine 15244 [running]:
github.com/Davincible/gotiktoklive.routineErrHandler({0xc002e7c1f0?, 0x3c?, 0xc0016cfec8?})
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/utils.go:330 +0x2d
github.com/Davincible/gotiktoklive.(*Live).readSocket(0xc002cef980)
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/wss.go:121 +0x305
created by github.com/Davincible/gotiktoklive.(*Live).tryConnectionUpgrade
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/wss.go:294 +0x136
```

**Comment by spinningsand** [comment]
> There is another thing I've noticed. Sometimes it only gets the audio while the video is a blank black screen.

Is there a recommended setting in VLC or another media player I should use that could help with this?

I don't know if this is useful, but I did find something else that lets you get stream info, but it's written in javascript. It gets the HLS stream instead of the FLV.  Could this could potentially help with playback issues?

https://github.com/Maxam0001/TikTokLiveStreamURLHelper

**Comment by luxcgo** [comment]
same issue, you're not using the latest version according to your log

**Comment by spinningsand** [comment]
Oops, I didn't see the latest update there.

**Comment by spinningsand** [comment]
Nevermind, it's working now. I extracted it to a different folder and it didn't extract right for some reason, so I just put it back in the same folder that I've been doing it in.

**Comment by spinningsand** [comment]
I'm still getting some blank videos that only has audio. Do you know anything that might be causing that?

**Comment by luxcgo** [comment]
Probably u could try a different parser and see where it goes. Download `ffmpeg` locally and add a line `Parser = "ffmpeg"` under  `[[Shows]]`
Btw, I'm only downloading the streams provided, not doing the fixing job. If stream itself is broken, there's not much I can do.

**Comment by spinningsand** [comment]
Thanks, the added ffmpeg line seems to have worked on every live that I've tested it on so far. 

I keep getting panics, I've deleted every old folder I had and re-installed the latest version:

```
panic: Failed to read websocket message, atempting to reconnect: read tcp 192.168.0.7:53623->104.76.210.216:443: wsarecv: An existing connection was forcibly closed by the remote host.

goroutine 325 [running]:
github.com/Davincible/gotiktoklive.routineErrHandler({0xc002f3a260?, 0x3c?, 0xc00293fec8?})
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/utils.go:330 +0x2d
github.com/Davincible/gotiktoklive.(*Live).readSocket(0xc000320080)
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/wss.go:121 +0x305
created by github.com/Davincible/gotiktoklive.(*Live).tryConnectionUpgrade
        C:/Users/CLUNKYBOXCOMPUTER/go/pkg/mod/github.com/!davincible/gotiktoklive@v0.0.0-20220525135242-dda499f27e6d/wss.go:294 +0x136
```

I deleted the !davincible folder along with everything related to olive before re-installing. When I check my files, I don't even have the !davincible folder anymore that the error is supposedly coming from.

**Comment by luxcgo** [comment]
I'm confused about how you installed it. If you build it from source, u need to run `go install github.com/go-olive/olive/src/cmd/olive@latest`.

**Comment by spinningsand** [comment]
I've been installing the .zip file. I deleted it again and re-installed again with the command line yesterday and I've been letting it run since then and so far, no issues right now. I'll keep letting it run and hopefully no more will pop up.  Thanks again for all your help! I'm new to programming so it's nice to have someone like you helping me out. I have it feeling it's gonna work good now..... hopefully lol....
